sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -1005,4 +1047,520 @@ export class DataLakeFileClient extends DataLakePathClient {
       span.end();
     }
   }
+
+
+  // high level functions
+
+  /**
+   * ONLY AVAILABLE IN NODE.JS RUNTIME.
+   *
+   * Uploads a local file to a Data Lake file.
+   *
+   * @param {string} filePath Full path of the local file
+   * @param {FileParallelUploadOptions} [options]
+   * @returns {(Promise<FileUploadResponse>)}
+   * @memberof DataLakeFileClient
+   */
+  public async uploadFile(
+    filePath: string,
+    options: FileParallelUploadOptions = {}
+  ): Promise<FileUploadResponse> {
+    const { span, spanOptions } = createSpan("DataLakeFileClient-uploadFile", options.tracingOptions);
+    try {
+      const size = (await fsStat(filePath)).size;
+
+      if (size > FILE_MAX_SIZE_BYTES) {
+        throw new RangeError(`size must be < ${FILE_MAX_SIZE_BYTES}.`);
+      }
+
+      // Create the file.
+      await this.create({
+        abortSignal: options.abortSignal,
+        metadata: options.metadata,
+        permissions: options.permissions,
+        umask: options.umask,
+        conditions: options.conditions,
+        pathHttpHeaders: options.pathHttpHeaders,
+        tracingOptions: { ...options!.tracingOptions, spanOptions }
+      });
+
+      // After the File is Create, Lease ID is the only valid request parameter.
+      options.conditions = { leaseId: options.conditions?.leaseId };
+
+      if (!options.chunkSize) {
```

Comment:
I remember max chunk size for flush operation is 100MB, please verify and add check here.

Reply:
See below. line 1093 - 1097.
```
      if (options.chunkSize <= 0 || options.chunkSize > FILE_UPLOAD_MAX_CHUNK_SIZE) {
        throw new RangeError(
          `chunkSize option must be > 0 and <= ${FILE_UPLOAD_MAX_CHUNK_SIZE}`
        );
      }
```